### PR TITLE
www-client/elinks fixes

### PR DESCRIPTION
**Pull request by doughdemon**

### CODE REVIEW

**Comment by thesamesam** [comment]
Ping. We can't move forward until the CI issue is resolved (see juippis' hint).

Do you need help doing this?

**Comment by doughdemon** [comment]
Thanks for the review. Please take a look at the updated commits.

File: www-client/elinks/elinks-0.13.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,138 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python2_7 )
```
Comment (thesamesam) [inline www-client/elinks/elinks-0.13.3.ebuild:5]: New line after EAPI

File: www-client/elinks/elinks-0.13.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+PYTHON_REQ_USE="threads"
+inherit autotools python-any-r1
```
Comment (thesamesam) [inline www-client/elinks/elinks-0.13.3.ebuild:7]: Newline before inherit

File: www-client/elinks/elinks-9999.ebuild [hunk 3]
```diff
@@ -41,6 +42,7 @@ DEPEND="${RDEPEND}
 
 PATCHES=(
 	"${FILESDIR}"/${P}-parallel-make.patch
+	"${FILESDIR}"/${PN}-0.13.3-ruby-gcc10.patch
 	)
```
Comment (thesamesam) [inline www-client/elinks/elinks-9999.ebuild:46]: Fix indentation for )

File: www-client/elinks/files/elinks-0.13.3-ruby-gcc10.patch [hunk 4]
```diff
@@ -0,0 +1,11 @@
+--- a/src/scripting/ruby/core.h
```
Comment (thesamesam) [inline www-client/elinks/files/elinks-0.13.3-ruby-gcc10.patch:1]: Is this upstreamed? Please add a link at the top of the file for where it came from.

File: www-client/elinks/elinks-9999.ebuild [hunk 5]
```diff
@@ -17,13 +17,14 @@ KEYWORDS=""
 IUSE="bittorrent brotli bzip2 debug finger ftp gopher gpm guile idn ipv6
 	javascript libressl lua +mouse nls nntp perl ruby samba ssl tre unicode X xml zlib"
 
+BDEPEND="virtual/pkgconfig"
 RDEPEND="
 	brotli? ( app-arch/brotli )
 	bzip2? ( >=app-arch/bzip2-1.0.2 )
 	gpm? ( >=sys-libs/ncurses-5.2:0= >=sys-libs/gpm-1.20.0-r5 )
 	guile? ( >=dev-scheme/guile-1.6.4-r1[deprecated] )
 	idn? ( net-dns/libidn )
```
Comment (thesamesam) [inline www-client/elinks/elinks-9999.ebuild:26]: Subslot dep (:=) on libidn if we link against it.

File: www-client/elinks/elinks-9999.ebuild [hunk 6]
```diff
@@ -17,13 +17,14 @@ KEYWORDS=""
...
-	javascript? ( >=dev-lang/spidermonkey-1.8.5:0= )
+	javascript? ( dev-lang/spidermonkey:17= )
```
Comment (thesamesam) [inline www-client/elinks/elinks-9999.ebuild:27]: :17 looks really old to me. Is this going to be improved or go away any time soon?

File: www-client/elinks/elinks-9999.ebuild [hunk 7]
```diff
@@ -17,13 +17,14 @@ KEYWORDS=""
 IUSE="bittorrent brotli bzip2 debug finger ftp gopher gpm guile idn ipv6
 	javascript libressl lua +mouse nls nntp perl ruby samba ssl tre unicode X xml zlib"
 
+BDEPEND="virtual/pkgconfig"
 RDEPEND="
 	brotli? ( app-arch/brotli )
```
Comment (thesamesam) [inline www-client/elinks/elinks-9999.ebuild:22]: If linking against brotli, := to rebuild when its ABI changes.

File: www-client/elinks/elinks-9999.ebuild [hunk 8]
```diff
@@ -1,7 +1,7 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
 PYTHON_COMPAT=( python2_7 )
```
Comment (thesamesam) [inline www-client/elinks/elinks-9999.ebuild:5]: Any chance of something newer than Python 2? There's a real push to get rid of Python 2 where possible.

File: www-client/elinks/elinks-0.13.3.ebuild [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+		$(use_enable xml xbel)
+	)
+
+	if use debug ; then
+		myconf+=( --enable-debug )
+	else
+		myconf+=( --enable-fastmem )
+	fi
+
+	# NOTE about GNUTSL SSL support (from the README -- 25/12/2002)
```
Comment (thesamesam) [inline www-client/elinks/elinks-0.13.3.ebuild:94]: Typo: GnuTLS, but is this still current?

File: profiles/arch/arm64/package.use.mask [hunk 10]
```diff
@@ -1,6 +1,10 @@
 # Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
+# Felix Janda <felix.janda@posteo.de> (2020-08-01)
+# Requires unkeyworded dev-lang/spidermonkey:17
+>=www-client/elinks-0.13.3 javascript
```
Comment (thesamesam) [inline profiles/arch/arm64/package.use.mask:6]: If we end up sticking with :17, remember to request keywording or at least investigate if :17 (which is old?) has support for these arches.

**Comment by doughdemon** [comment]
Thanks for the review, many of these are things I had planned to fix in
the future (and now got sped up).

> > @@ -0,0 +1,138 @@
> +# Copyright 1999-2020 Gentoo Authors
> +# Distributed under the terms of the GNU General Public License v2
> +
> +EAPI=7
> +PYTHON_COMPAT=( python2_7 )
> 
> New line after EAPI

fixed

> > @@ -0,0 +1,138 @@
> +# Copyright 1999-2020 Gentoo Authors
> +# Distributed under the terms of the GNU General Public License v2
> +
> +EAPI=7
> +PYTHON_COMPAT=( python2_7 )
> +PYTHON_REQ_USE="threads"
> +inherit autotools python-any-r1
> 
> Newline before inherit

fixed

> > @@ -41,6 +42,7 @@ DEPEND="${RDEPEND}
>  
>  PATCHES=(
>  	"${FILESDIR}"/${P}-parallel-make.patch
> +	"${FILESDIR}"/${PN}-0.13.3-ruby-gcc10.patch
>  	)
> 
> Fix indentation for )

fixed

> > @@ -0,0 +1,11 @@
> +--- a/src/scripting/ruby/core.h
> 
> Is this upstreamed? Please add a link at the top of the file for where it came from.

done

> > @@ -17,13 +17,14 @@ KEYWORDS=""
>  IUSE="bittorrent brotli bzip2 debug finger ftp gopher gpm guile idn ipv6
>  	javascript libressl lua +mouse nls nntp perl ruby samba ssl tre unicode X xml zlib"
>  
> +BDEPEND="virtual/pkgconfig"
>  RDEPEND="
>  	brotli? ( app-arch/brotli )
>  	bzip2? ( >=app-arch/bzip2-1.0.2 )
>  	gpm? ( >=sys-libs/ncurses-5.2:0= >=sys-libs/gpm-1.20.0-r5 )
>  	guile? ( >=dev-scheme/guile-1.6.4-r1[deprecated] )
>  	idn? ( net-dns/libidn )
> 
> Subslot dep (:=) on libidn if we link against it.
> 
> >  RDEPEND="
>  	brotli? ( app-arch/brotli )
>  	bzip2? ( >=app-arch/bzip2-1.0.2 )
>  	gpm? ( >=sys-libs/ncurses-5.2:0= >=sys-libs/gpm-1.20.0-r5 )
>  	guile? ( >=dev-scheme/guile-1.6.4-r1[deprecated] )
>  	idn? ( net-dns/libidn )
> -	javascript? ( >=dev-lang/spidermonkey-1.8.5:0= )
> +	javascript? ( dev-lang/spidermonkey:17= )
> 
> :17 looks really old to me. Is this going to be improved or go away any time soon?

that is currently a hard dependency, and not easy to fix since after
:17, it was switched to C++ whereas elinks is currently fully in C.
See https://github.com/rkd77/felinks/issues/29 .

> > @@ -17,13 +17,14 @@ KEYWORDS=""
>  IUSE="bittorrent brotli bzip2 debug finger ftp gopher gpm guile idn ipv6
>  	javascript libressl lua +mouse nls nntp perl ruby samba ssl tre unicode X xml zlib"
>  
> +BDEPEND="virtual/pkgconfig"
>  RDEPEND="
>  	brotli? ( app-arch/brotli )
> 
> If linking against brotli, := to rebuild when its ABI changes.

fixed

> >  # Distributed under the terms of the GNU General Public License v2
>  
> -EAPI=6
> +EAPI=7
>  PYTHON_COMPAT=( python2_7 )
> 
> Any chance of something newer than Python 2? There's a real push to get rid of Python 2 where possible.

done

> > +		$(use_enable gopher)
> +		$(use_enable ipv6)
> +		$(use_enable mouse)
> +		$(use_enable nls)
> +		$(use_enable nntp)
> +		$(use_enable samba smb)
> +		$(use_enable xml xbel)
> +	)
> +
> +	if use debug ; then
> +		myconf+=( --enable-debug )
> +	else
> +		myconf+=( --enable-fastmem )
> +	fi
> +
> +	# NOTE about GNUTSL SSL support (from the README -- 25/12/2002)
> 
> Typo: GnuTLS, but is this still current?

added gnutls support

> > @@ -1,6 +1,10 @@
>  # Copyright 1999-2020 Gentoo Authors
>  # Distributed under the terms of the GNU General Public License v2
>  
> +# Felix Janda <felix.janda@posteo.de> (2020-08-01)
> +# Requires unkeyworded dev-lang/spidermonkey:17
> +>=www-client/elinks-0.13.3 javascript
> 
> If we end up sticking with :17, remember to request keywording or at least investigate if :17 (which is old?) has support for these arches.

See the bug in the commit message.

**Comment by thesamesam** [comment]
Thanks for your response. Can you see if Python 3.7, 3.8, 3.9 are all possible and if so, add them all, rather than just 3.8?

I'll go over this again soon and test, but I think we're looking good.

**Comment by doughdemon** [comment]
> Thanks for your response. Can you see if Python 3.7, 3.8, 3.9 are all possible and if so, add them all, rather than just 3.8?

Ok, I tested those three python versions, and updated the ebuild.

**Comment by thesamesam** [comment]
Nearly there:

```
 * QA Notice: Unrecognized configure options:
 *
 *      --without-lua
 *      --without-lua
```

**Comment by thesamesam** [comment]
Also, I've adjusted the commit messages to adhere to [GLEP 66](https://www.gentoo.org/glep/glep-0066.html#commit-messages):
```
pick 97ce07de993 www-client/elinks: bump to EAPI 7
pick e5df7103580 www-client/elinks: 9999 is now python3 only
pick d714f8f4458 www-client/elinks: add support for gnutls
pick 302a433ea71 www-client/elinks: bump to 0.13.4
pick dc787e6b52e www-client/elinks: fix compilation with -fno-common/gcc10
pick 08e11da1101 www-client/elinks: USE=javascript needs dev-lang/spidermonkey:17
```

**Comment by doughdemon** [comment]
> ```
>  * QA Notice: Unrecognized configure options:
>  *
>  *      --without-lua
>  *      --without-lua
> ```

should be fixed now, please check.

**Pull request closed without merging**